### PR TITLE
Add support for hardened AMIs on arm64

### PR DESCRIPTION
**Pull request by reedloden**
This is only for the newer AL2023 AMIs, not the old AL2 AMIs (which are deprecated).

Depends on gravitational/cloud-terraform#3920.

`e` companion -- gravitational/teleport.e#3058

Ref #36110.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,18 @@
...
+### New features
+
+#### FIPS now supported on ARM64
+
+Teleport 15 now provides FIPS-compliant Linux builds on ARM64. Users will now
+be able to run Teleport in FedRAMP/FIPS mode on ARM64.
+
+#### Hardened AMI container images supported on ARM64
+
+Teleport 15 now provides hardened container images for AWS on ARM64.
```
Comment (wadells) [inline CHANGELOG.md:14]:
```suggestion
#### Hardened AMI supported on ARM64

Teleport 15 now provides hardened AWS AMIs on ARM64.
```

It is confusing to conflate OCI container images and AMIs.

File: docs/pages/installation.mdx [hunk 2]
```diff
@@ -367,14 +367,15 @@ the Teleport installation by setting configuration variables in the
...
 
 | Image name | Edition | FIPS support | AMI Tags | Owner Account ID |
 | - | - | - | - | - |
-| `teleport-oss-(=teleport.version=)-$TIMESTAMP` | OSS | No | `TeleportVersion: (=teleport.version=)`, `TeleportEdition: oss`, `TeleportFipsEnabled: false` | 146628656107 |
-| `teleport-ent-(=teleport.version=)-$TIMESTAMP` | Enterprise | No | `TeleportVersion: (=teleport.version=)`, `TeleportEdition: ent`, `TeleportFipsEnabled: false` | 146628656107 |
-| `teleport-ent-(=teleport.version=)-fips-$TIMESTAMP` | Enterprise | Yes | `TeleportVersion: (=teleport.version=)`, `TeleportEdition: ent`, `TeleportFipsEnabled: true` | 146628656107 |
+| `teleport-oss-(=teleport.version=)-$ARCH-$TIMESTAMP` | OSS | No | `TeleportVersion: (=teleport.version=)`, `TeleportEdition: oss`, `TeleportFipsEnabled: false` | 146628656107 |
+| `teleport-ent-(=teleport.version=)-$ARCH-$TIMESTAMP` | Enterprise | No | `TeleportVersion: (=teleport.version=)`, `TeleportEdition: ent`, `TeleportFipsEnabled: false` | 146628656107 |
+| `teleport-ent-(=teleport.version=)-$ARCH-fips-$TIMESTAMP` | Enterprise | Yes | `TeleportVersion: (=teleport.version=)`, `TeleportEdition: ent`, `TeleportFipsEnabled: true` | 146628656107 |
```
Comment (wadells) [inline docs/pages/installation.mdx:378]: I'd probably expand this to all 6 rows instead of templating some things (arch) but not others (oss/ent and fips).

Should we include the architecture tag in the `AMI Tags` column?

I don't feel super strongly about it if you like the current format.

Comment (wadells) [reply docs/pages/installation.mdx:378]: Also -- this looks like a minorly breaking change, as patterns that would have matched the previous fips ami names no longer will.   Since ARCH immediately proceeds TIMESTAMP, the non-fips images will probably cheat in via a wildcard.

Basically, consider what https://github.com/gravitational/teleport/pull/36112/files#diff-5442869deeb8c3e1e670605f82d3ed05fc3b9db2873b483ff7efd636117baeefR103 would look like for consumers of these AMIs before and after.

Comment (reedloden) [reply docs/pages/installation.mdx:378]: Expanded to 6 rows, didn't include the tag (as it doesn't match the other formats), and updated changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -84,6 +95,29 @@ them elsewhere before July 2024. Following brownouts in May and June, Teleport
...
+and usability of Teleport-provided Amazon AMIs.
+
+##### Legacy Amazon Linux 2 AMIs
+
+Teleport-provided Amazon Linux 2 AMIs were deprecated, and Teleport 14 is the
+last version to produce such legacy AMIs. With Teleport 15's release, only
+the newer hardened Amazon Linux 2023 AMIs will be produced.
+
+The legacy AMIs will continue to be published for Teleport 13 and 14 throughout
+the remainder of these releases' lifecycle.
```
Comment (wadells) [inline CHANGELOG.md:110]: This change hasn't merged yet afaik. https://github.com/gravitational/teleport/issues/34282

I'd recommend we don't scope creep to include it herein, as there are other docs and example changes needed to complete the work. For instance:

```console
walt@vm:~/git/teleport$ rg 126027368216            
assets/aws/Makefile
44:AWS_ACCOUNT_ID ?= 126027368216

examples/aws/terraform/starter-cluster/README.md
98:# OSS: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss*'
99:# Enterprise: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent*'

examples/aws/terraform/starter-cluster/data.tf
28:  owners      = [126027368216]

examples/aws/terraform/ha-autoscale-cluster/README.md
46:# OSS: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss*'
47:# Enterprise: aws ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent*'

examples/aws/terraform/ha-autoscale-cluster/data.tf
6:  owners      = [126027368216]

docs/pages/deploy-a-cluster/deployments/aws-ha-autoscale-cluster-terraform.mdx
169:The AWS account ID which publishes these AMIs is `126027368216`. You can list the available AMIs with
178:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss-(=teleport.version=)'
183:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)'
188:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)-fips'

docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx
182:The AWS account ID which publishes these AMIs is `126027368216`. You can list the available AMIs with
191:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-oss-(=teleport.version=)'
196:  $ aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)'
201:  aws --region <Var name="us-west-2" /> ec2 describe-images --owners 126027368216 --filters 'Name=name,Values=gravitational-teleport-ami-ent-(=teleport.version=)-fips'
```

We can pick up the legacy AMI deprecation in the change that removes/fixes the above.

**Review by wadells — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]
